core/src/main/java/com/github/steanky/element/core/factory/BasicFactoryResolver.java

```diff
@@ -109,7 +111,7 @@ private static Key parseKey(final KeyParser parser, final @Subst(Constants.NAMES
                 if (typeArguments.length != 2) {
                     //this is likely unreachable, as we are guaranteed to be an instance of ElementFactory
                     throw elementException(elementClass,
-                            "Unexpected number of type arguments on FactoryMethod return type");
+                            "unexpected number of type arguments on FactoryMethod return type");
                 }
```

Comment:
if you say had a
FactoryImpl implements ElementFactory<A,B>
and the factory method returned FactoryImpl, does this still work out?

Reply:
good point, i'll see if i can inspect the ElementFactory superclass instead